moto/sqs/models.py

```diff
@@ -534,7 +534,12 @@ def add_message(self, message):
                 [backend.delete_message(self.name, m.receipt_handle) for m in messages]
             else:
                 # Make messages visible again
-                [m.change_visibility(visibility_timeout=0) for m in messages]
+                [
```

Comment:
What's the reason for this change?

Reply:
Otherwise it could now modify the returned message copy (the one with possibly filtered attributes) and the original message in the backend wouldn't be touched.

By using `backend.change_message_visibility` this can be avoided since it only references the receipt handle of the message and will do a lookup of the message in its internal `_messages` field. Previously this worked because these were always the same message objects.


This change was now required for this test to pass after the changes of this PR: 

https://github.com/spulec/moto/blob/be197caba6282260ca53112f1f5c7ab4a6675ca9/tests/test_awslambda/test_lambda_eventsourcemapping.py#L213-L274

Reply:
Gotcha, that makes sense